### ISSUE TITLE
Legend is not properly rendered when it is initially not visible

### ISSUE THREAD

**Issue by Chnoch** [amcharts/amcharts4#1157]
In case we have a non-default rendering of a legend which is during initial rendering hidden by CSS, the rendering can look broken when reenabling the visibility of the legend. The issue can be seen in the following Codepen, where the legend is made visible after 2 seconds:
https://codepen.io/chnoch/pen/BEKGgp

Is there anyway to force a rerendering of the legend? Or any other ideas how to avoid this issue?

**Comment by martynasma** [comment]
This should make the legend fix itself:

```
chart.legend.deepInvalidate();
```

We'll also discuss whether we can do that automatically.

**Comment by zeroin** [comment]
This is updated pen with some other minor fixes: https://codepen.io/team/amcharts/pen/jRqRbZ

**Comment by VietHo** [comment]
I've been having the same issue. With vuejs using multiple charts in multiple bootstrap-vue tab.

When I do a data/series update, only the charts of the current tab gets displayed correctly.

For the others the axes/legends are misplaced as if the rendering was done before some properties were set.

I've tried putting deepInvalidate() everywhere but it didn't work.

I've manage to make deepInvalidate work by triggering it on click on the chart.

I've ended up going with a full chart.dispose() and rerender everything on data change.
